pkg/apis/core/v1alpha1/portforward_types.go

```diff
@@ -0,0 +1,144 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource"
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource/resourcestrategy"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PortForward
+// +k8s:openapi-gen=true
+type PortForward struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PortForwardSpec   `json:"spec,omitempty"`
+	Status PortForwardStatus `json:"status,omitempty"`
+}
+
+// PortForwardList
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PortForwardList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PortForward `json:"items"`
+}
+
+// PortForwardSpec defines the desired state of PortForward
+type PortForwardSpec struct {
+	// The name of the pod to port forward to/from. Required.
+	Pod string `json:"pod,omitempty"`
+
+	// The namespace of the pod to port forward to/from. Defaults to the kubecontext default namespace.
+	//
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// The port on the Kubernetes pod to connect to.
+	ContainerPort int
+
+	// The port to expose on the current machine.
+	LocalPort int
+
+	// Optional host to bind to on the current machine (localhost by default)
+	//
+	// +optional
+	Host string
+}
+
+var _ resource.Object = &PortForward{}
+var _ resourcestrategy.Validater = &PortForward{}
+
+func (in *PortForward) GetObjectMeta() *metav1.ObjectMeta {
+	return &in.ObjectMeta
+}
+
+func (in *PortForward) NamespaceScoped() bool {
+	return false
+}
+
+func (in *PortForward) New() runtime.Object {
+	return &PortForward{}
+}
+
+func (in *PortForward) NewList() runtime.Object {
+	return &PortForwardList{}
+}
+
+func (in *PortForward) GetGroupVersionResource() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "tilt.dev",
+		Version:  "v1alpha1",
+		Resource: "portforwards",
+	}
+}
+
+func (in *PortForward) IsStorageVersion() bool {
+	return true
+}
+
+func (in *PortForward) Validate(ctx context.Context) field.ErrorList {
+	// TODO(user): Modify it, adding your API validation here.
+	// TODO(maia): verify that Pod is populated, ContainerPort and LocalPort are
```

Comment:
FYI whenever you add validation, you'll need to add a blessed sample here:
https://github.com/tilt-dev/tilt/blob/77ae37f8f66c4b2d6df5eb526bd71954be73ecf2/internal/hud/server/apiserver_test.go#L40-L49